access/types/forms.py

```diff
@@ -605,18 +606,15 @@ def row_options(self, configuration, row):
             opt.append(trgopt)
         return { 'options': opt }
 
-    def grade_checkbox(self, configuration, value, hints=None):
+    def grade_checkbox(self, configuration, value, hints=None, name=''):
         hints = hints or []
         correct_count = 0
         wrong_answers = 0
         non_neutral_count = 0
-        # hack: configuration is modified in the method add_field so that
-        # the random_sample can be easily used here
-        randomized_sample = configuration.get('random_sample')
-        is_randomized = bool(randomized_sample)
-        sample = []
-        if is_randomized:
-            sample = [int(x) for x in randomized_sample.split('-')]
+        # Non-random checkbox questions have empty string as random_sample.
+        randomized_sample = self.fields[name].random_sample if name else ''
+        sample = [int(x) for x in randomized_sample.split('-')]
```

Comment:
If an empty string is split, it returns a list with one element, the empty string. Calling `int('')` raises `ValueError`. Did you test this with normal, non-random questions?

Furthermore, note that `bool([""]) == True` and `bool([]) == False`.